cli-plugins/socket/socket.go

```diff
@@ -35,16 +38,26 @@ func randomID() string {
 	return hex.EncodeToString(b)
 }
 
-func accept(listener *net.UnixListener, conn **net.UnixConn) {
+// accept creates a new Unix socket connection
+// and sends it to the *net.UnixConn channel
+// it allows reconnects
+func accept(listener *net.UnixListener) <-chan *net.UnixConn {
+	connChan := make(chan *net.UnixConn, 1)
+
 	go func() {
-		for {
-			// ignore error here, if we failed to accept a connection,
-			// conn is nil and we fallback to previous behavior
-			*conn, _ = listener.AcceptUnix()
-			// perform any platform-specific actions on accept (e.g. unlink non-abstract sockets)
-			onAccept(*conn, listener)
-		}
+		// this is a blocking call and will wait
+		// until a new connection is accepted
+		// or until the timout is reached
+		conn, _ := listener.AcceptUnix()
+
+		// perform any platform-specific actions on accept (e.g. unlink non-abstract sockets)
+		onAccept(listener)
+		connChan <- conn
+		// close the channel to signal we won't accept any more connections
+		close(connChan)
```

Comment:
nit: maybe we can `defer` this `close` as the very first line of the goroutine, just in case some panic happens somewhere else and we want to be sure to clean up the resources. WDYT?

Reply:
yeah that makes sense to me